src/pages/UpdateProfile/UpdateProfile.tsx

```diff
@@ -0,0 +1,217 @@
+import Button from "../../components/Button/Button";
+import "./UpdateProfile.scss";
+import { UserProfile } from "../../types/User";
+import Header from "../../components/Header/Header";
+import Navigation from "../../components/Navigation/Navigation";
+import TextField from "@mui/material/TextField";
+import { ChangeEvent, useState } from "react";
+// import { Link } from "react-router-dom";
+import { doc, updateDoc } from "firebase/firestore";
+import { auth, db } from "../../firebase";
+import { signInWithEmailAndPassword, updatePassword } from "firebase/auth";
+import { Dialog, DialogContent, DialogActions } from "@mui/material";
+
+type UpdateProfileProps = {
+  currentUser: UserProfile;
+  setCurrentUser: (updatedUser: UserProfile) => void;
+};
+const UpdateProfile = ({ currentUser, setCurrentUser }: UpdateProfileProps) => {
+  const [user, setUser] = useState<UserProfile>(currentUser);
+  const [password, setPassword] = useState<string>("");
```

Comment:
I think you should group all the password states into an about similar to how you are handling the different user fields.

Reply:
I have combined all the passwords into 1 state and I have also clubbed together the logic for the password display field